packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -5297,764 +5333,885 @@ describe('NetworkController', () => {
+                    rpcUrl: 'https://mock-rpc-url-2',
+                    chainId: '0x222',
+                    ticker: 'TEST2',
+                    nickname: 'test network 2',
+                    rpcPrefs: {
+                      blockExplorerUrl: 'https://test-block-explorer-2.com',
+                    },
                   },
                 },
               },
+              infuraProjectId: 'some-infura-project-id',
             },
             async ({ controller }) => {
-              const fakeProvider = buildFakeProvider();
-              const fakeNetworkClient = buildFakeClient(fakeProvider);
-              createNetworkClientMock.mockReturnValue(fakeNetworkClient);
-              await controller.setActiveNetwork('testNetworkConfigurationId');
+              const fakeProviders = [buildFakeProvider(), buildFakeProvider()];
+              const fakeNetworkClients = [
+                buildFakeClient(fakeProviders[0]),
+                buildFakeClient(fakeProviders[1]),
+              ];
+              mockCreateNetworkClient()
+                .calledWith({
+                  rpcUrl: 'https://mock-rpc-url-2',
+                  chainId: '0x222',
+                  type: NetworkClientType.Custom,
+                })
+                .mockReturnValue(fakeNetworkClients[0])
+                .calledWith({
+                  network: networkType,
+                  infuraProjectId: 'some-infura-project-id',
+                  type: NetworkClientType.Infura,
+                })
+                .mockReturnValue(fakeNetworkClients[1]);
+              await controller.setActiveNetwork('testNetworkConfiguration');
               expect(controller.state.providerConfig).toStrictEqual({
-                ...customNetworkConfiguration,
-                rpcUrl: rpcUrlOrTarget,
-                type: NetworkType.rpc,
+                type: 'rpc',
+                id: 'testNetworkConfiguration',
+                rpcUrl: 'https://mock-rpc-url-2',
+                chainId: '0x222',
+                ticker: 'TEST2',
+                nickname: 'test network 2',
+                rpcPrefs: {
+                  blockExplorerUrl: 'https://test-block-explorer-2.com',
+                },
               });
 
               await controller.rollbackToPreviousProvider();
 
-              expect(controller.state.providerConfig).toStrictEqual(
-                initialProviderConfig,
-              );
+              expect(controller.state.providerConfig).toStrictEqual({
+                type: networkType,
+                chainId: '111',
+                rpcUrl: 'https://mock-rpc-url-1',
+                ticker: 'TEST1',
+                nickname: 'test network 1',
+                rpcPrefs: {
+                  blockExplorerUrl: 'https://test-block-explorer-1.com',
+                },
+              });
             },
           );
         });
 
-        it('emits NetworkController:providerConfigChange via the messenger', async () => {
```

Comment:
Nit: it seems that this was removed accidentally? (and same for the RPC version of this test). But we're about to remove it anyway, so it's not such a big deal

Reply:
I'll add it back for completeness, for now, until we remove it.